src/utils/status/vm/vmStatus.js

```diff
@@ -137,16 +127,22 @@ const isWaitingForVmi = vm => {
   return NOT_HANDLED;
 };
 
-export const getVmStatus = (vm, launcherPod, importerPods, migration) =>
-  isBeingMigrated(vm, migration) || // must be precceding isRunning() since vm.status.ready is true for a migrating VM
-  isRunning(vm) ||
-  isReady(vm) ||
-  isVmError(vm) ||
-  isCreated(vm, launcherPod) ||
-  isBeingImported(vm, importerPods) ||
-  isWaitingForVmi(vm) || { status: VM_STATUS_UNKNOWN };
+export const getVmStatus = (vm, pods, migrations, importerPods = null) => {
```

Comment:
note: it is now possible to pass pods without any filtering to get the correct vm status. ImporterPods can be passed as an optimization which is up to the consumer